core/src/main/java/org/infinispan/notifications/cachelistener/cluster/ClusterListenerReplicateCallable.java

```diff
@@ -99,8 +97,7 @@ public Void call() throws Exception {
                                                                            cacheManagerNotifier, eventManager, sync);
                   // We only need to use the post cache event filter when a filter is provided, since the remote cluster listener
                   // will ignore pre events anyways.  This is mostly so a user filter is not notified of the pre event.
-                  cacheNotifier.addListener(listener,
```

Comment:
Ouch. Just remembered why I removed that ... It's because a KeyValueFilterConverter gets wrapped in a CacheEventFilter so the 'filter and convert in one step' aspect gets lost. So I need to remove the CompositeCacheEventFilter/PostCacheEventFilter wrapping but I still do not want to have my filter notified of pre events. :)